#407 Update UI Overview (#409)
Inbibo/adonisfx_docs: docs/houdini/ui_overview.md

## Adonis TAB Menu

> The Adonis nodes catalog can be inspected in the TAB Menu inside any geometry context under the submenu *Adonis*. It allows for quick access and creation of Adonis SOPs and HDAs. All of them are distributed in 5 submenus by type: Deformers, Locators, Sensors, Solvers, Tools and Utils.

Author: Fixed.

<figure style="width: 50%;" markdown>
  ![Adonis TAB menu](images/ui_overview_tab_menu.png)
  <figcaption><b>Figure 1</b>: Adonis TAB Menu inside a geometry context.</figcaption>
</figure>

| Icon | Description | TAB Submenu |
| :--- | :---------- | :---------- |

> | ![AdnClosestFit](../images/adn_closest_fit.png) | Creates an AdnClosestFit SOP. A deformer used to snap each point of the input mesh to the closest point on the target surface. | Adonis > Deformers > *AdnClosestFit* |

Reviewer: Just one small question, was it agreed that we would not document the inverse LBS node?
CC: @carlosmonteagudoinbibo @josevalverdeinbibo @albertochimentiinbibo

Author: We said that we would not be documenting it for now. That's why we haven't created the corresponding pages.

We can decide what to do now.

Reviewer 2: I don't remember tbh, but I think we don't need to document it. Let's not add it here, and we confirm with Marco later in the scrum. To me, it is an auxiliary node, not meant to be used in rigs by users. Anyways, let's ask Marco later

Reviewer: I think we should not document this

> | ![AdnMLDeformer](../images/adn_ml_deformer.png) | Creates an AdnMLDeformer SOP. A deformer used to infer mesh deformations from joint animations based on a trained machine learning model. | Adonis > Deformers > *AdnMLDeformer* |

Reviewer: I was thinking thast maybe we should add an extra column in which we mention in which context the nodes will be available, For example SOP or TOP. It might be easier for the user to read. I let @carlosmonteagudoinbibo decide

Author: Currently we have to tables. One for the SOP context and another one for the TOP Networks. Do you mean having a single table where we add a "Context"/"Type" column or something like that?

Reviewer 2: I am ok with two tables. But it's true that having one table and a 3rd column with the context would be cool too.

Reviewer: This is not a must, so if it's too convoluted let's leave it

Reviewer 3: please feel free to proceed with this or not, approved in both cases for me.

Author: We agreed to avoid adding a new column as the tables are already quite large.

| ![AdnMush](../images/adn_relax.png) | Creates an AdnMush SOP. A deformer used to smooth out the surface while preserving the details. | Adonis > Deformers > *AdnMush* |
| ![AdnPush](../images/adn_push.png) | Creates an AdnPush SOP. A deformer that pushes the geometry surface along the normal direction. | Adonis > Deformers > *AdnPush* |
| ![AdnRadialWrap](../images/adn_relax.png) | Creates an AdnRadialWrap SOP. A deformer used to transfer the topological details from a goal mesh to an input mesh based on landmark pairs. | Adonis > Deformers > *AdnRadialWrap* |
| ![AdnRelax](../images/adn_relax.png) | Creates an AdnRelax SOP. A deformer used to smooth creases and correct over-compression or over-stretching on geometry. | Adonis > Deformers > *AdnRelax* |
| ![AdnRigidWrap](../images/adn_relax.png) | Creates an AdnRigidWrap SOP. A deformer used to attach each point of the input mesh to its closest target point using rigid transformations. | Adonis > Deformers > *AdnRigidWrap* |

> | ![AdnSoftWrap](../images/adn_relax.png) | Creates an AdnSoftWrap SOP. A deformer used to deform the input mesh using weighted influences from nearby target points within a defined radius | Adonis > Deformers > *AdnSoftWrap* |

Author: Fixed.

| ![AdnSimshape](../images/adn_simshape.png) | Creates an AdnSimshape SOP. Solver for facial simulation. | Adonis > Solvers > *AdnSimshape* |
| ![AdnSkin](../images/adn_skin.png) | Creates an AdnSkin SOP. Solver for fascia and skin simulation. | Adonis > Solvers > *AdnSkin* |
| ![AdnSkinMerge](../images/adn_skin_merge.png) | Creates an AdnSkinMerge SOP. Node used to blend animation and simulation skin layers. | Adonis > Solvers > *AdnSkinMerge* |

> | ![AdnSmartTissue](../images/adn_smart_tissue.png) | Creates an AdnSkinMerge SOP. | Adonis > Solvers > *AdnSmartTissue* |

Author: Fixed.

> | ![AdnSmartTissue](../images/adn_smart_tissue.png) | Creates an AdnSmartTissue SOP. | Adonis > Solvers > *AdnSmartTissue* |

Author: I would need a bit of help for a short description of AdnSmartTissue. It seems we give a short explanation of the node in the Houdini page but not in the Maya page.

cc @carlosmonteagudoinbibo @lucasferryinbibo @albertochimentiinbibo

Reviewer: This would be my take:

Solver for adding simulation to already animated geometry by converting it into a dense tetrahedral volume, using a push-in parameter to control the inward offset, and modulating tissue properties with machine-learning data.

Reviewer 2: ```suggestion
| ![AdnSmartTissue](../images/adn_smart_tissue.png) | Creates an AdnSmartTissue SOP. Solver that applies skin dynamics without requiring the simulation of internal anatomy layers. It also enables stiffness to be modulated across the surface using an Adonis ML-trained model that emulates the activation of underlying muscles. | Adonis > Solvers > *AdnSmartTissue* |
```

Author: Done.

|||

> | ![AdnMLDataProcessing](../images/adn_data_extraction_tool.png) | Creates an AdnMLDataProcessing HDA. Node used to extract the deformation from the render skin and define the joints to be used in the data extraction process. | Adonis > Tools > *AdnMLDataProcessing* |

Reviewer 2: Let's add it here too

```suggestion
| ![AdnMLDataProcessing](../images/adn_data_extraction_tool.png) | Creates an AdnMLDataProcessing HDA. Node used to extract the deformation from the render skin and define the joints to be used in the data extraction process. An Adonis ML license is needed to use this HDA. | Adonis > Tools > *AdnMLDataProcessing* |
```

Author: Done.

> | ![AdnNeuralClusteringPaintTool](../images/adn_neural_clustering_paint_tool.png) | Creates an AdnSkinMerge HDA. Node used to paint the neural clusters needed for the training. An Adonis ML license is required to use this HDA. | Adonis > Tools > *AdnNeuralClusteringPaintTool* |

Author: Fixed.

> | ![AdnMLDataProcessing](../images/adn_data_extraction_tool.png) | Creates an AdnMLDataProcessing HDA. Node used to extract the deformation from the render skin and define the joints to be used in the data extraction process. An Adonis ML license is needed to use this HDA. | Adonis > Tools > *AdnMLDataProcessing* |

Reviewer 2: Just noticed, is the icon correct here?

Author: We are using the same icon for AdnMLDataExtraction and AdnMLDataProcessing.

- **Utils > Create Muscle PieceID**. Creates a `connectivity` node for each SOP in the selection in charge of computing the primitive attribute `path` that will identify each muscle piece.

> - **Utils > Install ML Dependencies**. Installs the Python dependencies required for inference with *AdnMLDeformer* and *AdnSmartTissue*, as well as for training with *AdnMLTraining*.

Reviewer 2: ```suggestion
- **Utils > Install ML Dependencies**. Installs the Python dependencies required to run Adonis ML training, and also to run the ML inference in *AdnMLDeformer* and *AdnSmartTissue* on the GPU.
```

Apply this also in maya please

Author: Done.

- **Turbo**. Opens the Turbo UI, which allows users to build an Adonis rig on a clean asset from scratch. The UI is divided into sections for each simulation layer that the AdnTurbo can configure. Users can toggle layers on or off to include or skip them in the execution and select the scene objects required to create and configure the solvers.

> - **Transfer**. Opens the AdnTransfer UI, which allows users to transfer the anatomy of muscles, fascia, fat, and skin cut geometries after morphing the mummy and skin geometries using AdnRadialWrap.

Reviewer 2: Can you standardize one thing please?
Sometimes we refer to a node in italic format, sometimes with plain text. Let's use the same rule always. I would maybe go with using italic always?

```suggestion
- **Transfer**. Opens the AdnTransfer UI, which allows users to transfer the anatomy of muscles, fascia, fat, and skin cut geometries after morphing the mummy and skin geometries using *AdnRadialWrap*.
```

Author: Done.

### License section

- **Activate License**. Checks the license status and if it is not activated yet, then a dialog will be prompted to guide on the product key registration. This functionality is only available in the Interactive Node-Locked license.

> - **Deactivate License**. Checks the license status and if it is activated, a dialog will be prompted asking confirmation before closing Maya. This functionality is only available if the license is Node-Locked.

Reviewer 2: ```suggestion
- **Deactivate License**. Checks the license status and if it is activated, a dialog will be prompted asking confirmation before closing Houdini. This functionality is only available if the license is Node-Locked.
```

Author: I thought I fixed this, but the DCC names were actually swapped. Fixed now.

> - **Deactivate License**. Check the license status and if it is activated, a dialog will be prompted asking confirmation before closing Maya. This functionality is only available in the Interactive Node-Locked license.

Author: Fixed.

> - **Deactivate License**. Checks the license status and if it is activated, a dialog will be prompted asking confirmation before closing Maya. This functionality is only available in the Interactive Node-Locked license.

Reviewer 2: ```suggestion
- **Deactivate License**. Checks the license status and if it is activated, a dialog will be prompted asking confirmation before closing Houdini. This functionality is only available if the license is Node-Locked.
```

Author: Done.

### Help section

- **Documentation**. Opens the Adonis technical documentation on a web browser.
- **Tutorials**. Opens the Adonis tutorials on YouTube on a web browser.
- **About**. Launches the Adonis About dialog with version information and credits.

> [!NOTE]

> > An Adonis ML license is required to use the following nodes: *AdnNeuralClusteringPaintTool*, *AdnMLDataExtraction* and *AdnMLTraining*.

Reviewer 2: ```suggestion
> An Adonis ML license is required to use *AdnNeuralClusteringPaintTool*, *AdnMLDataExtraction* and *AdnMLTraining*.
```

Author: Done.
